### PR TITLE
Issue 3286 replace deprecated aslist in web app generator test

### LINKED ISSUE #3286: WebAppGeneratorTest :  Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Task description

In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/e1cde0bd0d4ab1a659db7b1f7a884705048b27de/jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java#L129

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by ShivamChavan01**
## Description

Fixes #3286: Replace deprecated asList() with asInstanceOf(InstanceOfAssertFactories.list(type.class)) in WebAppGeneratorTest

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3291 (_2024-07-31T19:01:23Z_)

:gear: [JKube E2E Tests (10186086199)](https://github.com/eclipse-jkube/ci/actions/runs/10186086199) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3291)](https://github.com/eclipse-jkube/ci/actions/runs/10186086199/job/28177093310)

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 1]
```diff
@@ -172,7 +175,7 @@ void withOverriddenProperties_shouldAddImageConfiguration() throws IOException {
           .extracting(BuildConfiguration::getAssembly)
           .hasFieldOrPropertyWithValue("excludeFinalOutputArtifact", true)
           .hasFieldOrPropertyWithValue("user", "root")
-          .extracting("inline.files").asList().extracting("destName")
+          .extracting("inline.files").asInstanceOf(InstanceOfAssertFactories.list(Type.class)).extracting("testName")
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:178]: typo
```suggestion
          .extracting("inline.files").asInstanceOf(InstanceOfAssertFactories.list(Type.class)).extracting("destName")
```

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java [hunk 2]
```diff
@@ -126,7 +129,7 @@ void withDefaultHandler_shouldAddImageConfiguration() throws IOException {
             }})
           .extracting(BuildConfiguration::getAssembly)
           .hasFieldOrPropertyWithValue("excludeFinalOutputArtifact", true)
-          .extracting("inline.files").asList().extracting("destName")
+          .extracting("inline.files").asInstanceOf(InstanceOfAssertFactories.list(Type.class)).extracting("destName")
```
Comment (rohanKanojia) [inline jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:132]: You need to check what's the type of field in [AssemblyConfiguration](https://github.com/eclipse-jkube/jkube/blob/master/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/AssemblyConfiguration.java#L100)'s `inline` field, then check what's type of `files` inside `inline` (`Assembly`)

https://github.com/eclipse-jkube/jkube/blob/e1cde0bd0d4ab1a659db7b1f7a884705048b27de/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/Assembly.java#L53

You should use this type in `InstanceOfAssertFactories.list()` call

File: jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java (hunk 1 above)
Comment (ShivamChavan01) [reply jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java:178]: I thought it is a "TestName" I'll Change it

**Comment by rohanKanojia** [comment]
@ShivamChavan01 : Could you please update this pull request to contain only files related to #3286

**Comment by ShivamChavan01** [comment]
Sure I'll do it

**Comment by ShivamChavan01** [comment]
Made the changes Please Review @rohanKanojia

**Pull request closed without merging**